source/run-java-queries.txt

```diff
@@ -0,0 +1,75 @@
+.. _intellij-run-java-queries:
+
+================
+Run Java Queries
+================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+The {+intellij-full+} enables you to run queries written in Java directly 
+in the Database Explorer Playgrounds. 
+
+The Run button appears next to your MongoDB queries.
+
+.. image:: /images/run-queries/intellij-run-icon.png
+   :alt: Click the Run button to convert your Java query to mongosh syntax.
+
+Behavior
+--------
+
+When you click the Run button, the plugin automatically converts your 
+Java query to ``mongosh`` syntax and opens a Playground file with the 
+populated query.
+
+For field values that are variables determined at runtime, the plugin 
+creates a placeholder variable. You can populate this placeholder with 
+a test value and run the query in the Playground. 
+
+Example
+-------
+
+In this example, the Java query on the ``production.trips`` collection
+resemles the following:
+
+.. code:: java
+   :copyable: false
+
+   public List<Document> findCompletedTripsByDriver(String driverId) {
+      return trips.find(Filters.and(
+              Filters.eq(fieldName: "trip_status", value: "completed"),
+              Filters.eq(fieldName: "driver_id", driverId)
+      )).into(new ArrayList<>());
+   }
+
+The code example below shows the converted query from the Java code 
+above.
+
+.. code:: javascript
+   :copyable: false
+
+   var driver_id = ""
+
+   db.getSiblingsDB("production")
+     .getCollection("trips")
+     .find( { 
+        "$and" : [ 
+           { "trips.status" : "completed" },
+           { "driver_id" : driver_id }
+        ]
+      )
+
+In this example, ``driver_id`` is a variable that holds a value 
+determined at runtime.
+
+Once you are satisfied with the query, you can run it in the Playground 
```

Comment:
Thinking a bit more - can we also include how to run the query in the Playground console here? I know it should be trivial but there are a couple of ways of running a query and this is the preferred method. 

<img width="811" alt="Screenshot 2024-11-21 at 5 23 20 PM" src="https://github.com/user-attachments/assets/57920563-a40e-4ff3-be51-bcc450d44e5a">

Reply:
I've included this. Thanks!